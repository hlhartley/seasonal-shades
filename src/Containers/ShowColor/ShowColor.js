import React, { Component } from 'react';
import ColorCard from '../../Containers/ColorCard/ColorCard';
import { connect } from 'react-redux';
import ProductCard from '../ProductCard/ProductCard';

class ShowColor extends Component {
    constructor() {
        super()
    }
    render() {
        const type = this.props.type.toUpperCase()
        if(this.props.allColors) {
            return(
                <div className='color-info-card'>
                    <h2 className='color-info-card-title'>{`${type}`}</h2>
                    <ColorCard color={this.props.color}/>
                    <div className='product-info-card'>
                        <ProductCard color={this.props.color}/>
                    </div>
                </div>
            )
        } else {
            return(
                'loading color...'
            )
        }
    }
}

const mapStateToProps = (state) => ({
    nail_polishColors: state.nailpolishColors,
    lipstickColors: state.lipstickColors,
    blushColors: state.blushColors,
    eyeshadowColors: state.eyeshadowColors,
    allColors: state.allColors,
});


export default connect(mapStateToProps)(ShowColor);
