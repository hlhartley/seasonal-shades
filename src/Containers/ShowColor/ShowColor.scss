.color-info-card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.product-info-card {
    align-items: center;
    background-color: #fff;
    display: flex;
    justify-content: space-evenly;
    margin: auto;
    margin-bottom: 20px;
    margin-top: 20px;
    text-align: center;
    width: 550px;
}

.color-info-card-title {
    margin-top: 40px;
    text-align: center;
}