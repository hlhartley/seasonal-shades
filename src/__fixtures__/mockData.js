export const allColors = {
    522: {
        product: {},
        hexcode: '#EED8BE'
    },
    771: {
        product: {},
        hexcode: '#BD9E9B' 
    },
        'casablanca': {
        product: {},
        hexcode: '#444446'
    }
};

export const mockState = {
    allColors: {
        522: {
            product: {},
            hexcode: '#EED8BE'
        },
        771: {
            product: {},
            hexcode: '#BD9E9B' 
        },
        'casablanca': {
            product: {},
            hexcode: '#444446'
        }
    },
    favorites: {
        '001lune': true,
        '001petal': true,
    },
    blushColors: undefined,
    eyeshadowColors: undefined,
    lipstickColors: [],
    nail_polishColors: undefined,
    moreColors: undefined,
};

export const mockFavorites = {
    '001lune': true,
    '001petal': true, 
}