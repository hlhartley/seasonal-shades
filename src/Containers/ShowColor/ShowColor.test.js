import React from 'react'
import ShowColor, { mapStateToProps } from './ShowColor'
import { shallow } from 'enzyme';
import { allColors, mockState } from '../../__fixtures__/mockData';

describe('ShowColor', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<ShowColor color={'Casablanca'} type={'lipstick'} allColors={allColors} lipstickColors={[]} blushColors={[]} eyeshadowColors={[]} nail_polishColors={[]}/>)
    });

    describe('ShowColor component', () => {
        it('should match the snapshot with all the correct data passed in', () => {
            expect(wrapper).toMatchSnapshot()
        })
    });

    describe('mapStateToProps', () => {
        it('should return an object with all colors', () => {
            const expected = {
                allColors: {
                    522: {
                        product: {},
                        hexcode: '#EED8BE'
                    },
                    771: {
                        product: {},
                        hexcode: '#BD9E9B' 
                    },
                    'casablanca': {
                        product: {},
                        hexcode: '#444446'
                    }
                },
                blushColors: undefined,
                eyeshadowColors: undefined,
                lipstickColors: [],
                nail_polishColors: undefined,
            }
            const mappedProps = mapStateToProps(mockState)
            expect(mappedProps).toEqual(expected)
        });
    });
});