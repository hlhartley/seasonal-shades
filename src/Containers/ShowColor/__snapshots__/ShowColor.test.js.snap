// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowColor ShowColor component should match the snapshot with all the correct data passed in 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
